Update author list

richsalz opened the pull request:
This list was auto-generated from our database of everyone who agreed to the license change.
It is WIP until we close out the search.

levitte commented:
Hmmm, are you sure you want email addresses in this file?

richsalz commented:
Not positive, that's one of the reasons why it's marked as WIP.
But look at https://chromium.googlesource.com/chromium/src.git/+/master/AUTHORS and https://golang.org/AUTHORS  Google is an excellent source of best practices.  See also https://hg.mozilla.org/mozilla-central/file/tip/AUTHORS

paulidale commented on AUTHORS:
```diff
@@ -1,21 +1,631 @@
+Paul Kehrer <paul.l.kehrer@gmail.com>
+Paul Yang <paulyang.inf@gmail.com>
+<Paul.Green@stratus.com>
+Pauli <paul.dale@oracle.com>
```
Dr. Paul Dale <paul.dale@oracle.com>

paulidale commented on AUTHORS:
```diff
@@ -1,21 +1,631 @@
+Dr Stephen Henson <shenson@drh-consultancy.co.uk>
+Dr. Lutz Janicke <lutz@lutz-jaenicke.de>
+Dr. Matthias St. Pierre <Matthias.St.Pierre@ncp-e.com>
+Dr. Stephen Henson <steve@openssl.org>
```
Two Dr. Stephen Hensons ?

levitte replied:
Three, actually.  `<drh@celocom.com>` (below this comment) is him as well, and if you grep for `shenson@`, you'll find yet a few more a bunch of lines down.

richsalz replied:
There could be a zillion; they all should stay for reasons I described.

levitte commented:
@paulidale makes a good point, the file needs to be edited for duplicates

richsalz commented:
No, @paulidale is wrong, for a couple of reasons.  First, people have different employers; the work I did at Zolera, IBM, DataPower for example, is covered by different agreements. Second, how do you come up with a single address for someone? Third, how can you be sure that Brian Smith is the same Brian Smith everywhere?

levitte commented:
> First, people have different employers; the work I did at Zolera, IBM, DataPower for example, is covered by different agreements.

So you're seeing your authorships as different roles depending on what domain was on your email address?  You and I don't see it the same way then (but then, I have stubbornly refused to give away my day to day OpenSSL work to any company I've worked for, the project itself being the obvious exception)...  also, if you look at the moral rights (something that I believe is seen as more important in my part of the world, but I might be wrong), they always belong with the *person* regardless of working place, not with the *role* Name @ Company.

> Second, how do you come up with a single address for someone?

I don't, but this is one reason I asked you earlier if you actually wanted addresses in this file.

However, I could very well imagine having a list of addresses following my name:

```
Richard Levitte <levitte@lp.se>, <levitte@stacken.kth.se>, <levitte@openssl.org>, <richard@levitte.org>, <richard@openssl.com>, <richard@opensslfoundation.com>
```

*At a minimum*, I'd like not to be spread out all over the file.

> Third, how can you be sure that Brian Smith is the same Brian Smith everywhere?

I can't.  So that might be a reason to keep email addresses around.  In cases of doubt, let's try asking...  (I already gave you the answer re me)

richsalz commented:
I've done enough asking to get permission for the license change.  I am not going to go through the author list and ask about merging possible duplicates.  Feel free.

The EU moral right doesn't apply to the the US and probably other places, and it is far from clear that it applies to employees of some corporations.

There is nothing *wrong* with keeping the list with possible duplicates, and there is the risk of getting things wrong.

levitte commented:
Ok.  I'm *asking you* to at least group my diverse identities together.  Doesn't have to be all on one line of you're opposed to it, but it would be good, for clarity, if they were all prefixed with my name.

richsalz commented:
Post a diff.

I really really REALLY do not want to do work on this.  I don't want us to take on handling every special case.  It's an automated list, at least for now, generated by commit logs.  Will someone create a post-commit trigger for future authors?  If not, then this has to stay automated.

richsalz commented:
Then how will this file be maintained if not automatically?  The only alternative I can think of is a commit-hook (or whatever) that compares authors vs the AUTHORS file.  Will you write that?  Until we have a mechanism for making sure that we don't omit anyone, the script stands as-is.  So sorry, @levitte and sorry @paulidale no customizations, it's too risky to get wrong.

As I said when I created this PR.  It' WIP not because I expect edits, but because the final list of AUTHOR's isn't done yet.

Again, if someone automates this, then my concerns go away.

levitte commented:
In the FLOSS projects I know, it's a voluntary and author drivven thing, i.e. someone asks to be added.

[Google expresses it like this](https://opensource.google.com/docs/releasing/authors/):

> It is not necessary to proactively list the copyright holder for every single contribution you receive. Rather, it is sufficient to add copyright holders when they request to be added.

davidben commented:
Perhaps:
1. Parse the existing file into (name, email) pairs.
2. Generate (name, email) pairs automatically.
3. Merge the two lists by email address. If an email address appears in both lists and both entries have a name, use the name in the file, to allow for manual fixes.

That should still make sure entries aren't omitted while allowing for cases where names need to be fixed up. (Or perhaps if we later discover that some entry should be there but isn't captured by the script for whatever reason.)

t-j-h commented:
Putting a place holder -1 on this so it must go to an OMC vote. The whole strategy here isn't one I think makes sense.

richsalz commented:
Well f---.  This was discussed a long time and months ago.  Fine.

richsalz commented:
Based on https://opensource.google.com/docs/releasing/authors/, I have changed this to be a corporate list of copyright holders and added a comment.  Taking this out of WIP.

levitte commented:
Would it make sense to separate corporations and people into a list each (in the same file, but still)?

I'm not saying it has to happen in *this* PR, just asking the question.

richsalz commented:
Read the link and see what you think.  Google separates contributors from copyright-holding authors, which makes sense to me.

mspncp commented on AUTHORS:
```diff
@@ -1,631 +1,42 @@
+LLC Samsung R&D Institute Russia
+Levitte Programming
+Miracl Limited
+NCP Engineeing GmbH
```
NCP engineering GmbH

richsalz replied:
fixed. updated commit pushed.  thanks.

mspncp replied:
Are you sure you pushed it?

richsalz commented:
Of COURSE I am sure.  :)  I mean, I just ... double-checked... and, unh, oops.

levitte commented:
> Read the link and see what you think. Google separates contributors from copyright-holding authors, which makes sense to me.

I read the link, and the NOTE in there indicates that Google separate*D* (past tense) them into two files, but do not any longer.

Contributors are also copyright-holding (your way of expressing it might be interpreted differently)

iamamoose commented:
Wouldn't we want the list of corporate copyright holders to be in a file called NOTICE?  That way when we move to the Apache License it requires derivatives to preserve those statements.   http://www.apache.org/dev/licensing-howto.html#mod-notice

richsalz commented:
I thought Apache's NOTICE file was for required third-party copyrights that carry a separate license or requirements and we don't have that.  And @levitte sure, we can just add individuals to the AUTHORS file when requested to.  Either this PR or a separate one.

levitte commented on AUTHORS:
```diff
@@ -1,21 +1,65 @@
+# Individuals
+Andy Polyakov
+Ben Laurie
+Bodo Moeller
```
Missed one!  To copy-n-paste: Bodo Möller

richsalz commented:
Fixed all of them, thank you!

levitte commented on AUTHORS:
```diff
@@ -1,21 +1,65 @@
+Oracle America Inc.
+Rambus, Inc.
+Red Hat, Inc.
+Ribose, Inc.
```
I know another candidate that should be in here...

```
: ; git grep -n -C1 'Royal Institute'
CHANGES-5231-  *) Add processing of proxy certificates (see RFC 3820).  This work was
CHANGES:5232:     sponsored by KTH (The Royal Institute of Technology in Stockholm) and
CHANGES-5233-     EGEE (Enabling Grids for E-science in Europe).
--
crypto/x509v3/v3_pci.c-14- * Copyright (c) 2004 Kungliga Tekniska Högskolan
crypto/x509v3/v3_pci.c:15: * (Royal Institute of Technology, Stockholm, Sweden).
crypto/x509v3/v3_pci.c-16- * All rights reserved.
--
crypto/x509v3/v3_pcia.c-14- * Copyright (c) 2004 Kungliga Tekniska Högskolan
crypto/x509v3/v3_pcia.c:15: * (Royal Institute of Technology, Stockholm, Sweden).
crypto/x509v3/v3_pcia.c-16- * All rights reserved.
```

levitte replied:
The Swedish name is "Kungliga Tekniska Högskolan", translated to english as "Royal Institute of Technology, Stockholm, Sweden".  Choose the one you like, it doesn't matter as far as I know...

richsalz commented:
Added Swedish Institute.  (All these will be squashed once merged)  Thanks!

levitte commented:
"Add Royalty"  :-D :-D

mspncp commented:
For this pull request it would probably be appropriate if it would be approved by another @openssl/omc member, right?

richsalz commented:
Yes, we'll talk about this out our F2F.

t-j-h commented on README:
```diff
@@ -1,9 +1,12 @@
- Copyright (c) 1998-2018 The OpenSSL Project
- Copyright (c) 1995-1998 Eric A. Young, Tim J. Hudson
- All rights reserved.
+ Copyright 2016-2018 The OpenSSL Project Authors. All Rights Reserved.
```
That year is wrong - it should be 1998- ... to match usage elsewhere ...

richsalz replied:
fixed, additional commit pushed.

richsalz commented:
Will open a new PR at some point, closing this.
